mountpoint-s3/tests/reftests/reference.rs

```diff
@@ -110,6 +125,31 @@ impl Reference {
             };
         }
     }
+
+    /// Get a node from a full path, if it exists
+    pub fn lookup(&self, path: &str) -> Option<&Node> {
+        let path = Path::new(path);
+
+        let mut components = path.components().peekable();
+        assert_eq!(components.next(), Some(Component::RootDir));
+
+        let mut node = &self.root;
+        for dir in components {
+            node = match node {
+                Node::Directory(children) => {
+                    let dir = dir.as_os_str().to_str().unwrap().to_string();
+                    children.get(&dir)?
+                }
+                _ => panic!("unexpected internal file node"),
```

Comment:
If I understand correctly, `lookup("/a/b")` panics if `a` is a file, but returns `None` if `a` is a directory (and `b` does not exist). 

Is that correct? Why not return `None` in both cases?

Reply:
The only usage is to lookup a file in an existing directory, so it works out there. Still maybe this is not a general method?

Reply:
Yeah, it should just return None here, I think. In the future we might want to be able to distinguish "a directory along the way didn't exist" from "the leaf didn't exist", but not right now.